03-components.Rmd

```diff
@@ -1,10 +1,129 @@
+
+### Data is publicly available
+
+A transparent analysis has data that is publicly shared so that others can re-run the analysis as you have. Data should be provided in a way that it can be programmatically accessed (downloaded by a script). Data also need to be well documented in the form of metadata.
+
+<div class = "warning">
+Data sharing is a critical piece for promoting the open sourceness of your analysis, however this often needs to be balanced with privacy if you work with human data or samples. These data will likely contain personal identifiable information (PII) and protected health information (PHI). For more details on this, we encourage you to see this [course about data management](https://jhudatascience.org/Ethical_Data_Handling_for_Cancer_Research/data-privacy.html).
+</div>
+
+It's imperative that you protect human data but that doesn't mean that your analysis cannot be publicly shared! These are not mutually exclusive goals, but will take a bit of thoughtful planning. In the upcoming chapters we will provide additional ideas and information for how you can conduct an open source analysis while appropriately protecting sensitive data.
+
+### Readable code
+
+Readable code is much more important than clever code. If you are the only one who knows what your code is doing, it will not only be difficult for other to contribute or vet your analysis, but in the future, you will probably not understand what your code is doing either.
+
+[Read this course chapter from the ITCR training network about how to write durable code ](https://jhudatascience.org/Reproducibility_in_Cancer_Informatics/writing-durable-code.html).
+
+### Well-documented
+
+A well-documented analysis is a reproducible analysis. If analyses didn't require a lot of decisions and human comprehension than documentation would be unnecessary -- but also a lot of data analysts would be out of a job because robots would be able to do it! Analysts and developers often think of documentation as an after thought, but good documentation is being actively developed along with the code. Arguably, it is more important to have clear documentation than even working code, because if broken code is well documented, others may be able to help make suggestions for how it can be fixed.
+
+Good documentation not only describes what happened in an analysis, but why it happened -- why did the analyst choose this method or parameter as opposed to others? Was there an additional analysis, literature, or other resource that led us to this conclusion? Documentation should describe not only what is happening, but the thought process that led us here.
+
+### Version controlled
+
+A reproducible analysis is a version controlled analysis. Analyses go through many iterations, side quests, and sometimes meet dead ends -- and this is okay, it is how data science works! -- but if not done properly with version control, can lead to an unruly code base and a lot of confused team members.
```

Comment:
If not defined earlier  - would want to define "version controlled"

Reply:
dictionary box could be a helpful reminder though even if it is defined earlier

Reply:
Adding a dictionary box! For this and for GitHub :+1: